kDriveCore/Data/Cache/DriveFileManager/DriveFileManager.swift

```diff
@@ -1029,24 +1029,18 @@ public final class DriveFileManager {
     }
 
     @discardableResult
-    public func createOrRemoveShareLink(for file: ProxyFile, right: ShareLinkPermission) async throws -> ShareLink? {
-        if right == .restricted {
-            // Remove share link
-            let response = try await removeShareLink(for: file)
-            if response {
-                return nil
-            } else {
-                throw DriveError.serverError
-            }
-        } else {
-            // Update share link
-            let shareLink = try await createShareLink(for: file)
-            return shareLink
+    public func createOrUpdateShareLink(frozenFile: File, right: ShareLinkPermission) async throws -> ShareLink? {
```

Comment:
I need to check how the UI articulates to remove the share link. I'm not sure there is one now.